src/index.ts

```diff
@@ -242,25 +389,55 @@ export function register (opts: RegisterOptions = {}): Register {
  * Create TypeScript compiler instance.
  */
 export function create (options: CreateOptions = {}): Register {
+  return createInternal(new OptionsHelper({
+    explicitOptions: options
+  }))
+}
+
+function createInternal (optionsHelper: OptionsHelper): Register {
+  let options = optionsHelper.merge()
   const ignoreDiagnostics = [
     6059, // "'rootDir' is expected to contain all source files."
     18002, // "The 'files' list in config file is empty."
     18003, // "No inputs were found in config file."
     ...(options.ignoreDiagnostics || [])
   ].map(Number)
 
-  const ignore = options.skipIgnore ? [] : (options.ignore || ['/node_modules/']).map(str => new RegExp(str))
-
   // Require the TypeScript compiler and configuration.
-  const cwd = options.dir ? resolve(options.dir) : process.cwd()
-  const isScoped = options.scope ? (fileName: string) => relative(cwd, fileName).charAt(0) !== '.' : () => true
+
+  /**
+   * Compute options that must be computed before *and* after loading tsconfig
+   * They are required to successfully parse tsconfig, but might be changed by
+   * ts-node options specified in the config file.
+   */
+  function recomputedOptions () {
+    const cwd = options.dir ? resolve(options.dir) : process.cwd()
+    const isScoped = options.scope ? (fileName: string) => relative(cwd, fileName).charAt(0) !== '.' : () => true
+    const compiler = require.resolve(options.compiler || 'typescript', { paths: [cwd, __dirname] })
+    const ts: typeof _ts = require(compiler)
+    const readFile = options.readFile || ts.sys.readFile
+    const fileExists = options.fileExists || ts.sys.fileExists
+    return { cwd, isScoped, compiler, ts, readFile, fileExists }
+  }
+
+  // compute enough options to read the config file
+  let { cwd, isScoped, compiler, ts, fileExists, readFile } = recomputedOptions()
+
+  // Read config file
+  const { config, options: tsconfigOptions } = readConfig(cwd, ts, fileExists, readFile, options)
```

Comment:
We could be explicit about `options` being passed to `readConfig` here. One feature I think is broken due to the current design is the `files` option, that functionality could be moved into this function instead of `readConfig`.

Reply:
Good catch.  I think we need to set empty arrays for `files` and `includes` *within* `readConfig` because otherwise TypeScript's `parseJsonConfigFileContent` will do a bunch of globbing filesystem calls to discover the full list of files.  Does that sound right?